dozer/cogs/starboard.py

```diff
@@ -0,0 +1,366 @@
+"""Cog to post specific 'Hall of Fame' messages in a specific channel"""
+import logging
+import typing
+
+import discord
+from discord.ext.commands import guild_only, has_permissions
+
+from ._utils import *
+from .. import db
+
+MAX_EMBED = 1024
+DOZER_LOGGER = logging.getLogger('dozer')
+VIDEO_FORMATS = ['.mp4', '.mov', 'webm']
+
+
+async def is_cancelled(config, message, me, author=None):
+    """Determine if the message has cancellation reacts"""
+    if author is None:
+        author = message.author
+
+    for reaction in message.reactions:
+        if str(reaction) != config.cancel_emoji:
+            continue
+
+        users = await reaction.users().flatten()
+        if author in users or me in users:
+            await message.add_reaction(config.cancel_emoji)
+            return True
```

Comment:
We can also return `False` early if this isn't true, since there can only be one reaction with a given emoji.